python/ray/serve/backend_worker.py

```diff
@@ -68,10 +70,33 @@ async def __init__(self, backend_tag, replica_tag, init_args,
             if is_function:
                 _callable = backend
             else:
-                # This allows backends to define an async __init__ method
-                # (required for FastAPI backend definition).
-                _callable = backend.__new__(backend)
-                await sync_to_async(_callable.__init__)(*init_args)
+                # Retry with exponential backoff if ran into transient
+                # failures; Will throw and notify backend_state if
+                # constructor failed with no retries left
+                for current_try in range(REPLICA_CONSTRUCTOR_RETRY_COUNT + 1):
```

Comment:
It seems like we should probably do this retrying in the controller level instead of inside the replica given that we'll likely need retries there anyways and it'd be best to only have it in one place. Thoughts?

Reply:
That's probably a better idea with current setup now to introduce deployment level states and per-replica status reporting. We will need to keep track of each replica's retry count and make decision to abort if all replicas failed to start, while they're probably newly spawned replicas retrying, ex:

Target replica 10, we had total of 10 unique replicas failed 3 times without ever reaching to running state, that's when we can abort rather than spawning replica #11; But if we had 1 running replica, even we accumulated 10 total replicas failed to start, it shouldn't prevent us from spawning #11 since it has reached "RUNNING" state.